streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -859,16 +899,6 @@ void closeAndCleanUpTasks(final Collection<Task> activeTasks, final Collection<T
             closeTaskDirty(task);
         }
 
-        // TODO: change type to `StreamTask`
```

Comment:
This is for case 3) in the description: this is already in the `completeTaskCloseClean` so is a duplicated code.